### DISCUSSION TITLE
Making C-Y-D into a desktop clock

### DISCUSSION THREAD

**Discussion by scottchiefbaker** [witnessmenow/ESP32-Cheap-Yellow-Display#201] [Ideas]
I would love to see the CYD converted in to a nerdy desktop clock. Something along the lines of the [Mario Clock](https://www.instructables.com/Mario-Bros-Clock/) or something with changeable faces. Has anyone tackled anything like that? I'd be happy to contribute.

**Comment by ArifShariar** [comment]
Hey, I just bought CYD and was planning to turn it into a desktop clock. I have not started working on it yet. We could start working on it together.

**Comment by clach04** [comment]
@scottchiefbaker and @ArifShariar did you guys make any progress on this?

I have a working proof of concept written in MicroPython and using Nano GUI. Screenshot in https://github.com/peterhinch/micropython-font-to-py/pull/71#issuecomment-2644411639

![image](https://github.com/user-attachments/assets/35b7ef31-c6cc-44cb-906e-45bd17c37017)

Using a simple json file (format in-progress, partially described in https://github.com/clach04/cyd_clocks/issues/39#issuecomment-2644399265 and sample https://github.com/clach04/cyd_clocks/blob/main/theme.json). The format is rather similar to the yaml format that https://github.com/mathoudebine/turing-smart-screen-python/blob/main/res/themes/BigClock/theme.yaml has - so you can simulate and edit on a PC (note requires my branch) using https://github.com/clach04/turing-smart-screen-python/tree/issue_648_simulate_arbitrary_screen_size theme editor

caveats:

1. Wifi not always reliable - which I'm using for setting the time - be nice to remove that need https://github.com/clach04/cyd_clocks/issues/34
2. attaching a speaker leaks to odd noises, even when NOT using the speaker - https://github.com/witnessmenow/ESP32-Cheap-Yellow-Display/discussions/99#discussioncomment-12106163
3. still WIP
